File: 7-reverse-integer.go
func reverse(x int) int {
    if x == 0 {
        return 0
    }
    neg := (x < 0)
    if neg {
        x = -x
    }
    ret := 0
    for ; x > 0; {
        ret = ret * 10 + x % 10
        x = x / 10
    }
    if neg {
        ret = -ret
    }
    if ret > 2147483647 || ret < -2147483648 {
        ret = 0
    }
    return ret
}

File: bwu/math/7-reverse-integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x == 0:
            return 0
        
        neg = False
        if x < 0:
            neg = True
            x = -x
        
        res = 0
        while x:
            res = res * 10 + x % 10
            x /= 10
            
        if res > (2 ** 31 - 1):
            return 0
            
        if neg:
            return -res
        else:
            return res
